arviz/data/io_json.py

```diff
@@ -0,0 +1,56 @@
+"""Input and output support for data."""
+
+from .converters import convert_to_inference_data
+from .io_dict import from_dict
+
+try:
+    import ujson as json
+except ImportError:
+    # Can't find ujson using json
+    import json
+
+
+def from_json(filename):
+    """Initialize object from a json file.
+
+    Parameters
+    ----------
+    filename : str
+        location of json file
+
+    Returns
+    -------
+    InferenceData object
+    """
+    with open(filename, "rb") as file:
+        idata_dict = json.load(file)
+
+    return from_dict(**idata_dict, save_warmup=True)
+
+
+def to_json(data, filename, *, group="posterior", coords=None, dims=None):
+    """Save dataset as a json file.
+
+    WARNING: Only idempotent in case `data` is InferenceData
+
+    Parameters
+    ----------
+    data : InferenceData, or any object accepted by `convert_to_inference_data`
+        Object to be saved
+    filename : str
+        name or path of the file to load trace
+    group : str (optional)
+        In case `data` is not InferenceData, this is the group it will be saved to
+    coords : dict (optional)
+        See `convert_to_inference_data`
+    dims : dict (optional)
+        See `convert_to_inference_data`
+
+    Returns
+    -------
+    str
+        filename saved to
+    """
+    inference_data = convert_to_inference_data(data, group=group, coords=coords, dims=dims)
+    file_name = inference_data.to_json(filename)
+    return file_name
```

Comment:
Here, now that `to_netcdf` has already been around for a while, I would actually diverge from its behaviour. I think that having grooups, coords and dims here is very confusing, we have had several issues and questions about people undestanding that `group="prior"` means that only the prior is stored, similar thing for coords. 

I personally think we can force users to convert manually before calling `to_json` and have `to_json` accept only inferencedata objects. cc @ahartikainen @aloctavodia 

Otherwise, I would modify the docsting to be _extremely_ clear on this. All three parameters should be explicit in that they are ignored if the input is inferencedata and they should link to convert_to_inference_data

Reply:
I'm fine with this suggestion